python/paddle/fleet/base/fleet_base.py

```diff
@@ -329,12 +330,19 @@ def minimize(self,
 
         optimize_ops = []
         params_grads = []
+
         if meta_optimizer:
             optimize_ops, params_grads = meta_optimizer.minimize(
                 loss,
                 startup_program=startup_program,
                 parameter_list=parameter_list,
                 no_grad_set=no_grad_set)
+
+            default_program = paddle.default_main_program()
+
+            if id(default_program) != id(loss.block.program):
```

Comment:
not sure whether id() is the best choice

Reply:
by `StackOverFlow`